Swift/Tests/DatabaseTest.swift

```diff
@@ -1162,6 +1162,84 @@ class DatabaseTest: CBLTestCase {
         XCTAssertEqual(try colA.indexes(), ["index1", "index2", "index3", "index4", "index5"])
     }
     
+    func testFullTextIndexExpression() throws {
+        let colA = try self.db.createCollection(name: "colA")
+        
+        // Create FTS index:
+        let detailItem = FullTextIndexItem.property("passage")
+        let index2 = IndexBuilder.fullTextIndex(items: detailItem)
+        try colA.createIndex(withName: "passageIndex", config: index2)
+        
+        var doc = createDocument("doc1")
+        doc.setString("The boy said to the child, 'Mommy, I want a cat.'", forKey: "passage")
+        try colA.save(document: doc)
+        
+        doc = createDocument("doc2")
+        doc.setString("The mother replied 'No, you already have too many cats.'", forKey: "passage")
+        try colA.save(document: doc)
+        
+        let plainIndex = Expression.fullTextIndex("passageIndex")
+        let qualifiedIndex = Expression.fullTextIndex("passageIndex").from("colAa")
+        
+        var query = QueryBuilder
+            .select(SelectResult.expression(Meta.id))
+            .from(DataSource.collection(colA).as("colAa"))
+            .where(FullTextFunction.match(plainIndex, query: "cat"))
+        
+        var rows = try verifyQuery(query, block: { (n, r) in
+            XCTAssertEqual(r.string(at: 0), "doc\(n)")
+        })
+        XCTAssertEqual(rows, 2);
+        
+        query = QueryBuilder
+            .select(SelectResult.expression(Meta.id))
+            .from(DataSource.collection(colA).as("colAa"))
+            .where(FullTextFunction.match(qualifiedIndex, query: "cat"))
+        
+        rows = try verifyQuery(query, block: { (n, r) in
+            XCTAssertEqual(r.string(at: 0), "doc\(n)")
+        })
+        XCTAssertEqual(rows, 2);
+    }
+    
+    // TODO: https://issues.couchbase.com/browse/CBL-3994
+    func testFTSQueryWithJoin() throws {
```

Comment:
https://issues.couchbase.com/browse/CBL-3994 is fixed. Is TODO is still required?